pkg/metrics/aggregator.go

```diff
@@ -29,7 +29,7 @@ func NewMetricsAggregator(aggregationInterval time.Duration) *AdoptionMetricsAgg
 	collector := &AdoptionMetricsAggregator{
```

Comment:
It would be nice if `NewMetricsAggregator` could be private. But I get that you need to use it for test to reduce the aggregation interval. Could you maybe put that in a docstring comment on the func? Including "Don't use this outside of test!"